containers/haskell/README.md

```diff
@@ -0,0 +1,48 @@
+# Haskell (Community)
+
+## Summary
+
+[Haskell](https://www.haskell.org/) is an advanced, purely functional programming language
+
+This definition will hopefully get you going quickly with Haskell running as a remote container in vscode.
+
+| Metadata                    | Value                                                                        |
+|---------------------------- | -----------------------------------------------------------------------------|
+| *Contributors*              | [Stuart Pike](https://github.com/stuartpike), [Chuck Lantz](https://github.com/Chuxel), [Javier Neira](https://github.com/Chuxel), [eitsupi](https://github.com/eitsupi) |
+| *Categories*                | Community, Haskell |
+| *Definition type*           | Dockerfile |
+| *Works in Codespaces*       | Yes |
+| *Container host OS support* | Linux, macOS, Windows |
+| *Container OS*              | Debian |
+| *Languages, platforms*      | Haskell |
+
+
+## Using this definition
+
+While the definition itself works unmodified, you can select the version of Haskell the container uses by updating the `VARIANT` arg in the included `.devcontainer/devcontainer.json` file.
+
+```json
+# [Choice] Haskell version: 9, 8
+ARG VARIANT=8
```

Comment:
The way the choice option works is it actually sets the `build.args` property in devcontainer.json.  

```suggestion
"build": {
    "dockerfile": "Dockerfile",
    "args": {
        "VARIANT": "9"
    }
}
```

You can also add this to devcontainer.json with a note saying this is what it's for.  e.g.

```jsonc
{
    //...
    "build": {
        "dockerfile": "Dockerfile",
        "args": {
	     // Update 'VARIANT' to pick a Haskell version: 9, 8
	     "VARIANT": "8"
        }
    }
    //...
}
```